### DISCUSSION TITLE
Why is this not a pfSense package?

### DISCUSSION THREAD

**Discussion by HovKlan-DH** [jaredhendrickson13/pfsense-api#339] [Q&A] [answered]
As I understand, then we will need to manually reinstall this API after every pfSense version upgrade. Why is this the case - and by that I actually mean, why is this not a package in pfSense, as I assume this would be quite popular? Why have this hazzle?

My guess is, that Netgate has some concerns about its security and by then will not allow it - or that they are working on a similar future functionality and by then will not compete with themselves, by allowing this?

Or what is the reason? :-)

**Comment by jaredhendrickson13** [comment, marked as answer]
Great question! I don't want to go too much into detail about this at the moment. I will say that Netgate did reach out to me last year about potentially merging this project in and making it an official package; which I am all for. From those discussions came a few action items for both myself and Netgate that needed to be done before it could be merged in feasibly. Those are currently being worked on, but there is no timeframe or guarantee that it will happen. Hopefully!

**Comment by stevo11811** [comment]
I was about to post this same question, this package would be killer if it could survive updates, great work BTW.

**Comment by philippenc** [comment]
Hello, it would be fantastic news if you could integrate your work with pfSense to create an official package. The API is crucial for task automation, which is sorely lacking at the moment. Over a year ago, I reached out to you via email to inquire about the compatibility of your API with pfSense Plus, and I'm pleased to see that it now is. Thank you for the great work.
